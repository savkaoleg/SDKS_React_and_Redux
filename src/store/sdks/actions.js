import { SET_SDKS } from './types'

export function setSdks(payload) {
  return {
    type: SET_SDKS,
    payload
  }
}
