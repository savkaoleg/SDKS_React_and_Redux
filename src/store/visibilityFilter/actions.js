import { HANDLE_CHANGE, HANDLE_SEARCH_CHANGE } from './types'

export function handleChange(newType, newAditional) {
  return {
    type: HANDLE_CHANGE,
    newType,
    newAditional
  }
}

export function handleSearcChange(newType, searchQuery) {
  return {
    type: HANDLE_SEARCH_CHANGE,
    newType,
    searchQuery
  }
}
