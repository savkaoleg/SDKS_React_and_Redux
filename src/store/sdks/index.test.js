import expect from 'expect'
import reducer from './index'
import { setSdks } from './actions'

const initialState = [ ]
const data = [
  {
    'title': 'Accengage',
    'id': 'accengage',
    'tags': [
      'marketing-automation'
    ]
  },
  {
    'title': 'AdBuddiz',
    'id': 'adbuddiz',
    'tags': [
      'ad-network'
    ]
  }
]

describe('>>> sdks > reducer', ()=>{
  it('setSdks snapshot', () => {
    const next = reducer(initialState, setSdks(data))
    expect(next).toMatchSnapshot()
  })

  it('setSdks', () => {
    const next = reducer(initialState, setSdks(data))

    expect(next).toEqual(data)
  })
})
