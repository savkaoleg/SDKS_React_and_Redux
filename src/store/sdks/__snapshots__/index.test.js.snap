// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> sdks > reducer setSdks snapshot 1`] = `
Array [
  Object {
    "id": "accengage",
    "tags": Array [
      "marketing-automation",
    ],
    "title": "Accengage",
  },
  Object {
    "id": "adbuddiz",
    "tags": Array [
      "ad-network",
    ],
    "title": "AdBuddiz",
  },
]
`;
